code/game/machinery/embedded_controller/airlock_program.dm

```diff
@@ -38,6 +44,10 @@
 
 	if (istype(M, /obj/machinery/embedded_controller/radio/airlock))	//if our controller is an airlock controller than we can auto-init our tags
 		var/obj/machinery/embedded_controller/radio/airlock/controller = M
+		cycle_to_external_air = controller.cycle_to_external_air
+		if(cycle_to_external_air)
+			tag_pump_out_external = "[id_tag]_pump_out_external"
+			tag_pump_out_internal = "[id_tag]_pump_out_internal"
```

Comment:
So the X_pump_out_external are pumps outside the airlock on the potentially-planet? And X_pump_out_internal are inside the potentially-spaceship?

Reply:
Almost.  There is a second set of pumps to do the "pump out" operation.  An internal one is *inside the airlock* and an external one on the planet.

Reply:
So pump_out pumps are used when cycling to the outside, then? pump_in is cycling to internal? Or, can you just list the tags and where they go?

Reply:
### Setup for a "cycle_to_external_air" shuttle airlock:
- `X_pump_out_external` - A vent pump on the "outside" of the shuttle, connected by pipes to the pump_out_internal vent.
- `X_pump_out_internal` - A vent pump inside the airlock chamber, connected by pipes to the pump_out_external vent.
- `X_pump` - A vent pump inside the airlock chamber, connected by pipes to air canisters.
- `X_exterior_sensor` - Sensor on the "outside" of the shuttle.  Must set this tag manually on the controller and sensor, its not auto-populated onto controller.
#### Cycling to "outside"
1. Purge the chamber by draining thru X_pump into the canisters.
2. Fill the chamber by siphoning planetary air via X_pump_out_external thru the pipe and out X_pump_out_internal.
#### Cycling to "inside"
1. Purge the chamber by draining thru X_pump_out_internal thru the pipe and out X_pump_out_external
2. Fill the chamber from the canisters thru X_pump